core/domain/story_domain.py

```diff
@@ -0,0 +1,359 @@
+CMD_UPDATE_STORY_PROPERTY = 'update_story_property'
+
+
+class StoryChange(object):
+    """Domain object for changes made to story object."""
+    STORY_PROPERTIES = (
+        STORY_PROPERTY_TITLE, STORY_PROPERTY_DESCRIPTION,
+        STORY_PROPERTY_TOPIC, STORY_PROPERTY_NOTES,
+        STORY_PROPERTY_STORY_CONTENTS, STORY_PROPERTY_LANGUAGE_CODE)
+
+    def __init__(self, change_dict):
+        """Initialize a StoryChange object from a dict.
+
+        Args:
+            change_dict: dict. Represents a command. It should have a 'cmd'
+                key, and one or more other keys. The keys depend on what the
+                value for 'cmd' is. The possible values for 'cmd' are listed
+                below, together with the other keys in the dict:
+                - 'update story property' (with property_name, new_value
+                and old_value)
+
+        Raises:
+            Exception: The given change dict is not valid.
+        """
+        if 'cmd' not in change_dict:
+            raise Exception('Invalid change_dict: %s' % change_dict)
+        self.cmd = change_dict['cmd']
+
+        if self.cmd == CMD_UPDATE_STORY_PROPERTY:
+            if (change_dict['property_name'] in
+                    self.STORY_PROPERTIES):
+                self.property_name = change_dict['property_name']
+                self.new_value = change_dict['new_value']
+                self.old_value = change_dict.get('old_value')
+            else:
+                raise Exception('Invalid change_dict: %s' % change_dict)
+
+    def to_dict(self):
+        """Returns a dict representing StoryChange domain object.
+
+        Returns:
+            dict. A dict representing StoryChange instance.
+        """
+        return {
+            'cmd': self.cmd,
+            'property_name': self.property_name,
+            'new_value': self.new_value,
+            'old_value': self.old_value
+        }
+
+
+class StoryNode(object):
+    """Domain object describing a node in the exploration graph of a
+    story.
+    """
+
+    def __init__(
+            self, node_id, destination_nodes,
+            acquired_skill_ids, prerequisite_skill_ids,
+            annotations, exploration_id):
+        """Initializes a StoryNode domain object.
+
+        Args:
+            node_id: str. The unique id for each node.
+            destination_nodes: list(str). The list of destination node ids that
+                this node points to in the story graph.
+            acquired_skill_ids: list(str). The list of skill ids acquired by
+                the user on completion of the node.
+            prerequisite_skill_ids: list(str). The list of skill ids required
+                before starting a node.
+            annotations: str. Free-form annotations that a lesson implementer
+                can use to construct the exploration. It described the basic
+                theme or template of the story.
+            exploration_id: str. The valid exploration id that fits the
```

Comment:
str or None, right? (And explain the None case)

Reply:
Done.